### PR TITLE
Update tf-master

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

File: DeepSpeech.py [hunk 1]
```diff
@@ -800,39 +818,15 @@ def do_graph_freeze(output_file=None, output_node_names=None, variables_blacklis
...
-                        ]
-                        for e in default_empty:
-                            self.__dict__[e] = None
-
-                flags = TFLiteFlags()
-                tflite_convert._convert_model(flags)
-                os.unlink(temp_freeze)
+
+                converter = lite.TFLiteConverter(frozen_graph, input_tensors=inputs.values(), output_tensors=outputs.values())
+                tflite_model = converter.convert()
```
Comment (lissyx) [inline DeepSpeech.py]: This lacks quantization

**Comment by reuben** [comment]
I requested review on the tensorflow PR, not this one.

**Comment by reuben** [comment]
Modulo the docker build, tests are green.

File: Dockerfile [hunk 2]
```diff
@@ -60,7 +60,7 @@ RUN wget https://bootstrap.pypa.io/get-pip.py && \
 # >> START Configure Tensorflow Build
 
 # Clone TensoFlow from Mozilla repo
-RUN git clone https://github.com/mozilla/tensorflow/
+RUN git clone https://github.com/reuben/tensorflow/
```
Comment (lissyx) [inline Dockerfile]: I guess you will want to remove that before merging?

File: Dockerfile [hunk 3]
```diff
@@ -186,7 +186,7 @@ RUN cp /tensorflow/bazel-bin/native_client/generate_trie /DeepSpeech/native_clie
 
 # Install TensorFlow
 WORKDIR /DeepSpeech/
-RUN pip install tensorflow-gpu==1.12.0rc2
+RUN pip install tensorflow-gpu==1.12.0
```
Comment (lissyx) [inline Dockerfile]: But that should likely be fixed?

File: Dockerfile (hunk 2 above)
Comment (reuben) [reply Dockerfile]: Yes.

File: Dockerfile (hunk 3 above)
Comment (reuben) [reply Dockerfile]: No, it'll be broken regardless. 1.13 requires new versions of CUDA/cuDNN/NCCL/etc, so installing 1.13 or not doesn't make any difference.

Comment (lissyx) [reply Dockerfile]: Well, we can update that `FROM nvidia/cuda:9.0-cudnn7-devel-ubuntu16.04`

Comment (reuben) [reply Dockerfile]: Probably. Then we also have to update the apt-get package names/versions, etc. I haven't tested.

**Pull request merged**